### ISSUE TITLE
multi-event feature request

### ISSUE THREAD

**Issue by mrTsjolder** [tensorflow/tensorboard#1450]
Currently, when having multiple event files in some subdirectory of the log directory, it is assumed that these event files are to be appended to get one single graph. This is rather often not the case, however, which leads to a bunch of users ending up with a bunch of useless lines connecting the different runs, e.g. as illustrated [here](https://stackoverflow.com/questions/36798669/how-can-i-separate-runs-of-my-tensorflow-code-in-tensorboard).

Now, I wanted to conduct several experiments and obviously repeat the experiments (with the same parameters) to assure these results are reproducible. This means that I have different runs (with different parameters) that are repeated for the sake of reproducibility. As a good tensorboard user, I put each run in its own subdirectory, but in the end, I have a multi-coloured chaos of learning curves from which I can barely learn anything. It is mostly doable to compare the different runs, but the repetitions have completely different colours and make it almost impossible to see things.

What I would like to see now, is some option that allows to use a single directory to store repeated runs of a single experiment that does away with the connections between different runs in the plots. Ideally, this option would also allow to plot the mean curve with error bars (over all runs) rather than the separate events. 

I am not sure whether my explanation makes a lot of sense or how feasible my suggestions are, but it would be really helpful to plot multiple events in some subdirectory in a single colour.

**Comment by nfelt** [comment]
Thanks for the feature request.  We have some upcoming work to support multiple experiments within TensorBoard and also better display of runs in hyperparameter tuning (including repeats of the same parameter setting) that I think should address your use case - stay tuned.

cc @erzel @stephanwlee FYI

**Comment by erzel** [comment]
Thanks for the feature request. As Nick wrote, we're working on a plugin that will allow plotting repeats of the same training session. It allows you to compare training sessions with different parameters and to group sessions with the same parameters so that you can plot the progress of the training metrics of each session in the same graph. It also allows you to get an aggregation of the last metric value in each such session in the group of repeates (mean, max, min, median, etc.). Adding a curve plotting the progress of the mean, is a good idea; I'll make note of it, so we can prioritize accordingly.

**Comment by dgrahn** [comment]
@erzel Any idea when that plugin would be available? I'm using Strat K-fold to produce a number of runs for each model architecture. It would be great to group all of the folds into a single line.

**Comment by erzel** [comment]
Hi @dgrahn. We currently have an internal version of it running and our plan is to open-source it by end of December. I suppose using the same "tag" for a metric across all folds and then using the current 'scalar' plugin to the same metric from all folds in a single graph chart is not good-enough?

**Comment by dgrahn** [comment]
@erzel Is the scalar plugin just the scalar tab? It'd be nice to compare the means of the folds for different architectures.

**Comment by dgrahn** [comment]
DC open source it. I would definitely contribute.

- Pardon my brevity. I sent this from my phone.
